src/app/pages/applications/applications.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { WebSocketService } from '../../services/index';
+
+@Injectable({ providedIn: 'root'})
+export class ApplicationsService {
+
+  constructor(private ws: WebSocketService) {}
+
+  getPoolList() {
+    return this.ws.call('pool.query');
+  }
+
+  getKubernetesConfig() {
+    return this.ws.call('kubernetes.config');
+  }
+
+  getAllCatalogItems() {
+    return this.ws.call('catalog.query', [[], {"extra": {"item_details": true}}]);
+  }
+
+  getBindIPChoices() {
+    return this.ws.call('kubernetes.bindip_choices');
+  }
+
+  getDockerImages() {
+    return this.ws.call('docker.images.query');
+  }
+
+  getCatItems() {
+    return this.ws.call('catalog.items', ['OFFICIAL']);
+  }
+
+  getChartReleases(name?: string) {
+    if (name) {
+      return this.ws.call('chart.release.query', [[['name', '=', name]]]);  
+    }
+    return this.ws.call('chart.release.query');
+  }
+
+  setReplicaCount(name: string, count: number) {
+    return this.ws.call('chart.release.scale', [name, { replica_count: count}]);
+  }
+
+  getPodConsoleChoices(name: string) {
+    return this.ws.call('chart.release.pod_console_choices', [name]);
+  }
+
+  getInterfaces() {
+    return this.ws.call('interface.query');
```

Comment:
Is every interface a possible choice, @sonicaj ?

Or should we provide a interface_choices?

Reply:
I think we should, it should work as we use macvlan/bridge automatically but still better to have a separate method to ensure there is no issue. I'll update Dennis.